include/boost/decimal/detail/cmath/lgamma.hpp

```diff
@@ -0,0 +1,230 @@
+// Copyright 2023 Matt Borland
+// Copyright 2023 Christopher Kormanyos
+// Distributed under the Boost Software License, Version 1.0.
+// https://www.boost.org/LICENSE_1_0.txt
+
+#ifndef BOOST_DECIMAL_DETAIL_CMATH_LGAMMA_HPP
+#define BOOST_DECIMAL_DETAIL_CMATH_LGAMMA_HPP
+
+#include <array>
+#include <type_traits>
+
+#include <boost/decimal/fwd.hpp> // NOLINT(llvm-include-order)
+#include <boost/decimal/detail/type_traits.hpp>
+#include <boost/decimal/detail/concepts.hpp>
+#include <boost/decimal/numbers.hpp>
+
+namespace boost {
+namespace decimal {
+
+// Begin: Forward declarations
+template <BOOST_DECIMAL_DECIMAL_FLOATING_TYPE T>
+constexpr auto log(T x) noexcept -> std::enable_if_t<detail::is_decimal_floating_point_v<T>, T>;
```

Comment:
Do you need these forward declarations? You should be able to just include the relevant headers since you won't end up with a cycle.

Reply:
> Do you need these forward declarations?

I suppose they are not needed.